src/parsers.rs

```diff
@@ -135,7 +135,7 @@ named!(pub parse_f32<f32>,
       map_res!(map_res!(fdigit, str::from_utf8), FromStr::from_str));
 
 /// Parses a sequence of whitespace seperated u32s.
-named!(pub parse_u32s<Vec<u32> >, separated_list!(space, parse_u32));
+named!(pub parse_u32s<Vec<u32> >, separated_list!(space, complete!(parse_u32)));
```

Comment:
> Could you explain what's going on here

Due to change to digit mentioned above `parse_u32s(b"")` will not complain about the incomplete input instead of returning an empty Vec.

> and why parse_i32s does not need to be changed?

Because it internally uses your custom `sdigit` function, which doesn't complain on empty input.